include/highfive/bits/H5Utils.hpp

```diff
@@ -51,7 +51,7 @@ struct inspector {
     static constexpr size_t recursive_number_dimensions = number_dimensions;
 
     static std::array<size_t, recursive_number_dimensions> getDimensions(const type& /* val */) {
-        return std::array<size_t, recursive_number_dimensions>{};
+        return std::array<size_t, recursive_number_dimensions>();
```

Comment:
```suggestion
        return std::array<size_t, recursive_number_dimensions>();
```
I'd say `recursive_number_of_dimensions` or `recursive_dimensions` (IMO `getDimensions()` → `dimensions()` would also be nice. STL has `.size()`, and not `.getSize()` :) )

Reply:
hm,
`getDimensions` to follow DataSpace: `getDimensions`, `getNumberDimensions`

Reply:
Ok, consistency is good. `getNumberDimensions` seems like a bad name to me, though… could there also be `getStringDimensions`?

Reply:
Just saying that IMO, `getNumberDimensions` reads a little weird. `getNumberOfDimensions` would have been better. (I misread the former as the dimensions of things that are numbers due to the missing "of", hence my question about the dimensions of things that are "strings" :) )

TL;DR: `DataSpace` already has these methods, my lingual confusion should be ignored.

Reply:
I agree with your @alkino in the sense that also I find `dimension()` (no 's'?) and `size()` a lot better than `getSize()` and `getDimensions()`. If the overhaul involves more API changes I think such a change could be discussed, but at the same time renaming is not super urgent. One could also phase in the change by having a deprecation warning for the old API. 

Then, `getNumberDimensions` I also find completely confusing: I immediately start thinking that this must be something special that I do not know about. In comparions `getNumberOfDimensions` is clear, but quite long and verbose. 

So my vote would be for an API change (following STL):
* `getSize()` -> `size()`
* `getDimensions()` -> `dimension()`

and for an internal use of
* `recursive_number_dimensions` -> `recursive_dimension` or `dimension_recursive`
* `number_dimensions` -> `dimension`

Reply:
> `getDimensions()` -> `dimensions()`

Should keep the plural there, no? And yes, these would have to be added, and the older ones deprecated properly.

Reply:
@matz-e If they are added my confusion below should be taken into account. I think that what is being called `dimensions` here should be called `shape`

Reply:
Guys I don’t think we should at this point take care of prettifying the API. We have a lot of work already and dealing with deprecations and new API just for renaming... it might not be perfect but I at least is consistent.
Wrt internal api sure, we can add the "of" to leave no doubts.

Reply:
@tdegeus ah, yes… I agree, being closer to numpy would be less confusing overall.

@ferdonline I think the one "of" should be in there. Otherwise, as per my tl;dr:, consistency trumps other stuff for right now.